runtime/mainnet/src/config/proxy.rs

```diff
@@ -1,79 +1,112 @@
-use codec::{Decode, Encode, MaxEncodedLen};
 use frame_support::traits::InstanceFilter;
-use pop_runtime_common::proxy::{MaxPending, MaxProxies};
-use sp_runtime::RuntimeDebug;
+use pop_runtime_common::proxy::{MaxPending, MaxProxies, ProxyType};
 
 use crate::{
-	deposit, parameter_types, Balance, Balances, BlakeTwo256, Runtime, RuntimeCall, RuntimeEvent,
+	config::assets::TrustBackedAssetsCall, deposit, parameter_types, Balance, Balances,
+	BlakeTwo256, Runtime, RuntimeCall, RuntimeEvent,
 };
 
-/// The type used to represent the kinds of proxying allowed.
-// Mainnet will use this definition of ProxyType instead of the ones in
-// `pop-common` crates until `pallet-assets` is in runtime.
-// `ProxyType` in `pop-common` include Assets specific proxies which won't
-// make much sense in this runtime.
-#[derive(
-	Copy,
-	Clone,
-	Eq,
-	PartialEq,
-	Ord,
-	PartialOrd,
-	Encode,
-	Decode,
-	RuntimeDebug,
-	MaxEncodedLen,
-	scale_info::TypeInfo,
-)]
-pub enum ProxyType {
-	/// Fully permissioned proxy. Can execute any call on behalf of _proxied_.
-	Any,
-	/// Can execute any call that does not transfer funds or assets.
-	NonTransfer,
-	/// Proxy with the ability to reject time-delay proxy announcements.
-	CancelProxy,
-	/// Collator selection proxy. Can execute calls related to collator selection mechanism.
-	Collator,
-}
-impl Default for ProxyType {
-	fn default() -> Self {
-		Self::Any
-	}
-}
-
-impl ProxyType {
-	/// Defines proxies permission hierarchy.
-	// Example: A proxy that is not superset of another one won't be able to remove
-	// that proxy relationship
-	// src: https://github.com/paritytech/polkadot-sdk/blob/4cd07c56378291fddb9fceab3b508cf99034126a/substrate/frame/proxy/src/lib.rs#L802
-	pub fn is_superset(s: &ProxyType, o: &ProxyType) -> bool {
-		match (s, o) {
-			(x, y) if x == y => true,
-			(ProxyType::Any, _) => true,
-			(_, ProxyType::Any) => false,
-			(ProxyType::NonTransfer, ProxyType::Collator) => true,
-			_ => false,
-		}
-	}
+fn is_transfer_call(call: &RuntimeCall) -> bool {
+	matches!(
+		call,
+		RuntimeCall::Balances { .. } |
+			RuntimeCall::Assets { .. } |
+			RuntimeCall::Nfts { .. } |
+			RuntimeCall::Treasury { .. }
+	)
 }
 
 impl InstanceFilter<RuntimeCall> for ProxyType {
 	fn filter(&self, c: &RuntimeCall) -> bool {
 		match self {
 			ProxyType::Any => true,
-			ProxyType::NonTransfer => !matches!(c, RuntimeCall::Balances { .. }),
+			ProxyType::NonTransfer =>
+				!is_transfer_call(c) &&
+					// Wrapped transfer calls are filtered too.
+					matches!(
+						c,
+						RuntimeCall::Utility(pallet_utility::Call::batch { calls }) |
+						RuntimeCall::Utility(pallet_utility::Call::batch_all { calls })
+						if !calls.iter().any(|call| is_transfer_call(call))
+					) && matches!(
+					c,
+					RuntimeCall::Utility(pallet_utility::Call::as_derivative { call, .. })
+					if !is_transfer_call(call)
+				),
```

Comment:
`NonTransfer` can't dispatch calls that transfer funds, nor wrapped transfer funds.